marshmallow/fields.py

```diff
@@ -290,6 +294,8 @@ def deserialize(self, data, fields_dict, many=False, validators=None,
                     exception_class=UnmarshallingError,
                     strict=strict
                 )
+                if field_obj.dump_only:
```

Comment:
This should go before the call to `_call_and_store`. If the field dump-only, no need to do the work of deserializing the value and validating it.